quizapp/src/test/groovy/billennium/quizapp/UserITSpec.groovy

```diff
@@ -0,0 +1,39 @@
+package billennium.quizapp
+
+
+import billennium.quizapp.entity.User
+
+import billennium.quizapp.repository.UserRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment
+import org.springframework.boot.test.web.client.TestRestTemplate
+import spock.lang.Specification
+
+
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+class UserITSpec extends Specification {
+
+    @Autowired
+    TestRestTemplate restTemplate
+
+    @Autowired
+    UserRepository userRepository
+
+
+    def "Insert a User Entity and retrieve the resulting String via GET /user/1"() {
+        given: "A User Entity"
+        userRepository.save([
+                name: "newUser"
+        ] as User)
+
+        when: "GET /user/1"
+        def result = restTemplate.getForObject("/user/1", String.class)
```

Comment:
średnio mi sie podoba używania restTemplate w teście (widze takie rozwiązanie pierwszy raz). Do wywoływania controllerów w Spock służy bean MockMvc na którym można wywołać operacje HTTP.  Ta opcja pozwala także sprawdzić response.   Ogólnie droga jaką wybrałeś (test integracyjny) aby to przetestować jest ok ale test jest do poprawy. Tutaj też warto zastosować klasę ControllerConstants o której Ci wspominałem.

Reply:
a widzisz poznałeś coś nowego :)